bokeh/properties.py

```diff
@@ -837,7 +837,8 @@ class List(Seq):
 
     """
 
-    def __init__(self, item_type, default=[], help=None):
+    def __init__(self, item_type, default=None, help=None):
+        default = default or []
```

Comment:
I don't understand. The `default` argument represents the default value that this list prop (as attached to a HasProps class) will have, right? There is no *default* default list, or is there? If there is, should it not be documented or implemented differently?

Just for clarity, with the current way of writing it (before this PR):

    from bokeh.properties import HasProps, List, Int
    
    class Foo(HasProps):        
        spam = List(Int)
        eggs = List(Int)
    
    # Modify the default list instance, which is shared by all List properties
    # that did not specify `default` 
    Foo.spam._default.append(3)  
    
    foo1 = Foo()
    print(foo1.spam, foo1.eggs)  # -> [3] [3]

Granted, this is not likely to cause problems anytime soon, but I cannot imagine this is intentional. Although I might be missing a point.

Of all the "fixes" I did in this PR, there is I think one where the situation really led to a bug. The other situations seem currently pretty safe. However, as the code evolves it might at some point lead to very strange bugs. And that's easily avoided by simply never using empty dicts/lists as the default value for an argument.